src/writer/handler.h

```diff
@@ -65,11 +65,11 @@ class DirentHandler {
 
     virtual void start() = 0;
     virtual void stop() = 0;
-    Dirent* getUniqueDirent() {
-      if (!mp_uniqueDirent) {
-        mp_uniqueDirent = getDirent();
+    Dirent* getDirent() {
+      if (!mp_dirent) {
+        mp_dirent = createDirent();
       }
-      return mp_uniqueDirent;
+      return mp_dirent;
     }
     virtual std::unique_ptr<ContentProvider> getContentProvider() const = 0;
```

Comment:
This comment pertains to the commit message. Since the earlier comment was addressed via a fixup commit (which is perfectly fine), after squashing the title of the fixed-up commit will not match its content. Don't forget to edit the commit message.

Reply:
I've reword the commit. Now we introduce the `DirentHandler::createDirent` method.